src/System.Net.Http/src/System/Net/Http/Unix/ReaderWriterStream.cs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Threading;
+
+namespace System.Net.Http
+{
+    internal class ReaderWriterStream : MemoryStream
```

Comment:
I'm going to refrain from commenting on this implementation until the whole synchronous vs asynchronous thing is worked out.  I'm concerned how much synchronous blocking is being done in this implementation, especially with this deriving from MemoryStream on which operations are expected to be close to instantaneous, where ReadAsync/WriteAsync just call the Read/Write methods synchronously, etc.